src/scaffoldmaker/utils/eftfactory_tricubichermite.py

```diff
@@ -819,6 +819,180 @@ def createEftWedgeXi1Zero(self):
+
+        # zero cross derivative parameters
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS2, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS2DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [1, 2, 3, 4, 5, 6, 5, 6]
+        remapEftLocalNodes(eft, 6, ln_map)
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftWedgeCollapseXi2RightAtXi3One:  Failed to validate eft'
+        return eft
+
+    def createEftWedgeCollapseXi2LeftAtXi3One(self):
+        '''
+        Create a tricubic hermite element field for a wedge element, where xi2 collapsed on xi3 = 1 creating the
+        wedge elements. Top left nodes are collapsed.
+        :return: Element field template
+        '''
+        eft = self.createEftBasic()
+        setEftScaleFactorIds(eft, [1], [])
+
+        # remap parameters on xi3 = 1 before collapsing nodes
+        remapEftNodeValueLabel(eft, [7, 8], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS2, [1])])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D_DS2, [])
+
+        # zero cross derivative parameters
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS2, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS2DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [1, 2, 3, 4, 5, 6, 5, 6]
+        remapEftLocalNodes(eft, 6, ln_map)
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftWedgeCollapseXi2LeftAtXi3One:  Failed to validate eft'
+        return eft
+
+    def createEftWedgeCollapseXi1AtXi3One(self):
+        '''
+        Create a tricubic hermite element field for a wedge element, where xi1 collapsed on xi3 = 1.
+        :return: Element field template
+        '''
+        eft = self.createEftBasic()
+
+        # remap parameters on xi3 = 1 before collapsing nodes
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D_DS1, [])
+        remapEftNodeValueLabel(eft, [5, 7], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS1, [])])
+
+        # zero cross derivative parameters
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS2, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS2DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [1, 2, 3, 4, 5, 5, 6, 6]
+        remapEftLocalNodes(eft, 6, ln_map)
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftWedgeCollapseXi1AtXi3One:  Failed to validate eft'
+        return eft
```

Comment:
I think we need both"createEftWedgeCollapseXi1QuadrantAtXi2Zero(self, collapseNodes)" and "createEftWedgeCollapseXi1QuadrantAtXi2One(self, collapseNodes)" since they have different node numbering and mapping. So, I decided to define one function named "createEftWedgeCollapseXi1QuadrantAtXi2Plane(self, collapseNodes, Xi2=None)" which include 4 cases:

- Xi2=0 and collapseNodes = [1, 5]
- Xi2=0 and collapseNodes = [2, 6]
- Xi2=1 and collapseNodes = [3, 7]
- Xi2=1 and collapseNodes = [4, 8]

Reply:
I'm also defining "createEftWedgeCollapseXi2QuadrantAtXi3Plane(self, collapseNodes, Xi3=None)" which includes following cases:

- Xi3=0 and collapseNodes = [1, 2]
- Xi3=0 and collapseNodes = [3, 4]
- Xi3=1 and collapseNodes = [5, 6]
- Xi3=1 and collapseNodes = [7, 8]

Reply:
and the third one as : "createEftWedgeCollapseXi1QuadrantAtXi3Plane"

Reply:
So, now I replaced 3 functions for creating wedge elements which cover 12 cases.

Reply:
I don't think you need to mention AtXi3Plane, it's Xi2 which is being collapsed and the list of nodes say where.